### DISCUSSION TITLE
Addrmap alignment

### DISCUSSION THREAD

**Discussion by RasmusGOlsen** [SystemRDL/systemrdl-compiler#111] [General]
In the systemrdl-compiler documentation, there is mentioned a [STRICT_SELF_ALIGN]( https://systemrdl-compiler.readthedocs.io/en/latest/api/messages.html?highlight=align#systemrdl.warnings.STRICT_SELF_ALIGN
) warning. Is it possible to instruct the compiler to use this alignment mode when distributing addresses instead of manually specifying the alignment? If not could we consider restructuring the code so users can easily modify the address distribution so this can be done with a user-defined property or something similar?

**Comment by amykyta3** [comment]
I think what you're looking for is the `addressing` property. This property controls how to pack addresses if they are not explicitly assigned.
If you set it to `addressing=fullalign`, then inferred addresses will get packed according to the `STRICT_SELF_ALIGN` rules.
